### ISSUE TITLE
Permissions related: "Error trying to access settings files in /opt/bitnami/ghost/versions/4.9.4/content/settings/."

### ISSUE THREAD

**Issue by jeffreycunn** [bitnami/bitnami-docker-ghost#115]
I have permissions error happen when running docker-compose up

```
error trying to access settings files in /opt/bitnami/ghost/versions/4.9.4/content/settings/.
```
Full error read out below

**Steps to reproduce the issue:**

1. sudo docker-compose up

**Describe the results you received:**
```
Recreating ghost-compose_mariadb_1 ... done
Recreating ghost-compose_ghost_1   ... done
Attaching to ghost-compose_mariadb_1, ghost-compose_ghost_1
ghost_1    | ghost 15:17:36.10 
ghost_1    | ghost 15:17:36.11 Welcome to the Bitnami ghost container
ghost_1    | ghost 15:17:36.11 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-ghost
mariadb_1  | mariadb 15:17:35.25 
mariadb_1  | mariadb 15:17:35.25 Welcome to the Bitnami mariadb container
mariadb_1  | mariadb 15:17:35.26 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb_1  | mariadb 15:17:35.26 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb_1  | mariadb 15:17:35.26 
mariadb_1  | mariadb 15:17:35.26 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 15:17:35.32 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 15:17:35.32 INFO  ==> Initializing mariadb database
ghost_1    | 
ghost_1    | Error ID:
ghost_1    |     4fcc5f60-e7db-11eb-96cc-d30b0272df73
ghost_1    | 
ghost_1    | Error Code: 
ghost_1    |     EACCES
ghost_1    | 
ghost_1    | ----------------------------------------
ghost_1    | 
ghost_1    | InternalServerError: Error trying to access settings files in /opt/bitnami/ghost/versions/4.9.4/content/settings/.
ghost_1    |     at /opt/bitnami/ghost/versions/4.9.4/core/frontend/services/settings/ensure-settings.js:40:23
ghost_1    |     at tryCatcher (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/util.js:16:23)
ghost_1    |     at Promise._settlePromiseFromHandler (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/promise.js:547:31)
ghost_1    |     at Promise._settlePromise (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/promise.js:604:18)
ghost_1    |     at Promise._settlePromise0 (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/promise.js:649:10)
ghost_1    |     at Promise._settlePromises (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/promise.js:725:18)
ghost_1    |     at _drainQueueStep (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/async.js:93:12)
ghost_1    |     at _drainQueue (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/async.js:86:9)
ghost_1    |     at Async._drainQueues (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/async.js:102:5)
ghost_1    |     at Immediate.Async.drainQueues [as _onImmediate] (/opt/bitnami/ghost/versions/4.9.4/node_modules/bluebird/js/release/async.js:15:14)
ghost_1    |     at processImmediate (internal/timers.js:464:21)
ghost_1    | 
ghost_1    | Error: EACCES: permission denied, copyfile '/opt/bitnami/ghost/versions/4.9.4/core/frontend/services/settings/default-routes.yaml' -> '/opt/bitnami/ghost/versions/4.9.4/content/settings/routes.yaml'
ghost_1    | 
ghost_1    | [2021-07-18 15:17:48] WARN Ghost is shutting down
ghost_1    | [2021-07-18 15:17:48] WARN Ghost has shut down
ghost_1    | [2021-07-18 15:17:48] WARN Your site is now offline
ghost_1    | [2021-07-18 15:17:48] WARN Ghost was running for a few seconds
mariadb_1  | 2021-07-18 15:17:48 23 [Warning] Aborted connection 23 to db: 'bitnami_ghost' user: 'bn_ghost' host: '172.20.0.3' (Got an error reading communication packets)
mariadb_1  | 2021-07-18 15:17:48 19 [Warning] Aborted connection 19 to db: 'bitnami_ghost' user: 'bn_ghost' host: '172.20.0.3' (Got an error reading communication packets)
mariadb_1  | 2021-07-18 15:17:48 21 [Warning] Aborted connection 21 to db: 'bitnami_ghost' user: 'bn_ghost' host: '172.20.0.3' (Got an error reading communication packets)
mariadb_1  | 2021-07-18 15:17:48 22 [Warning] Aborted connection 22 to db: 'bitnami_ghost' user: 'bn_ghost' host: '172.20.0.3' (Got an error reading communication packets)
mariadb_1  | 2021-07-18 15:17:48 20 [Warning] Aborted connection 20 to db: 'bitnami_ghost' user: 'bn_ghost' host: '172.20.0.3' (Got an error reading communication packets)
ghost-compose_ghost_1 exited with code 0
^CGracefully stopping... (press Ctrl+C again to force)
Stopping ghost-compose_mariadb_1   ... done
```
**Describe the results you expected:**

I already attempted to change permissions of the the directory:

```
sudo chown 1001:1001 -R ./persist-ghost/
```

**Additional information you deem important (e.g. issue happens only occasionally):**

```
version: '2'
services:
  mariadb:
    image: docker.io/bitnami/mariadb:10.3
    volumes:
      - '/masked/persist-mariadb:/bitnami/mariadb'
    environment:
      - ALLOW_EMPTY_PASSWORD=no
      - MARIADB_ROOT_PASSWORD=masked
      - MARIADB_PASSWORD=masked
      - MARIADB_USER=bn_ghost
      - MARIADB_DATABASE=bitnami_ghost
  ghost:
    image: docker.io/bitnami/ghost:4
    ports:
      - '80:2368'
    volumes:
      - '/masked/persist-ghost:/bitnami/ghost'
      - '/opt/bitnami/'
    depends_on:
      - mariadb
    environment:
      - ALLOW_EMPTY_PASSWORD=no
      - GHOST_PASSWORD=masked
      - GHOST_DATABASE_HOST=mariadb
      - GHOST_DATABASE_PORT_NUMBER=3306
      - GHOST_DATABASE_USER=bn_ghost
      - GHOST_DATABASE_NAME=bitnami_ghost
      - GHOST_DATABASE_PASSWORD=masked

```

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:40 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:48 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

- Output of `docker info`:

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 6
  Running: 0
  Paused: 0
  Stopped: 6
 Images: 3
 Server Version: 20.10.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc version: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-1046-gcp
 Operating System: Ubuntu 18.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.843GiB
 Name: website-ghost-container-ubuntu
 ID: LNDO:MJI7:MNY3:MAYP:O6DQ:G2II:WXIF:L7NP:VE7P:AGOV:MVKE:RZYC
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019

```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

GCP VM

**Comment by javsalgar** [comment]
Hi,

It seems that you are using local volumes. Could you check that the permissions of the mounted volumes are open enough so the user 1001 can write in it? Normally, in this kind of cases, you may need to give write permissions to the `other` group.

**Comment by jeffreycunn** [comment]
> Hi,
> 
> It seems that you are using local volumes. Could you check that the permissions of the mounted volumes are open enough so the user 1001 can write in it? Normally, in this kind of cases, you may need to give write permissions to the `other` group.

I tried this:  

```
sudo chmod o+rwx ./persist-ghost/ 

drwxrwxrwx 2 1001 google-sudoers 4096 Jul 18 14:31 persist-ghost
drwxrwxrwx 3 1001 google-sudoers 4096 Jul 18 14:31 persist-mariadb
```

Afterwards, with empty directories, the first time I ran 

```
sudo docker-compose up
```

The terminal output was:

```
Starting ghost-compose_mariadb_1 ... done
Starting ghost-compose_ghost_1   ... done
Attaching to ghost-compose_mariadb_1, ghost-compose_ghost_1
ghost_1    | ghost 11:39:08.68 
ghost_1    | ghost 11:39:08.68 Welcome to the Bitnami ghost container
ghost_1    | ghost 11:39:08.68 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-ghost
ghost_1    | ghost 11:39:08.68 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-ghost/issues
ghost_1    | ghost 11:39:08.69 
mariadb_1  | mariadb 11:39:07.92 
mariadb_1  | mariadb 11:39:07.92 Welcome to the Bitnami mariadb container
mariadb_1  | mariadb 11:39:07.92 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb_1  | mariadb 11:39:07.93 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb_1  | mariadb 11:39:07.93 
mariadb_1  | mariadb 11:39:07.93 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 11:39:07.98 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 11:39:07.99 INFO  ==> Initializing mariadb database
mariadb_1  | mariadb 11:39:08.02 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1  | mariadb 11:39:08.02 INFO  ==> Setting user option
mariadb_1  | mariadb 11:39:08.04 INFO  ==> Installing database
ghost_1    | ghost 11:39:08.69 INFO  ==> Configuring libnss_wrapper
ghost_1    | ghost 11:39:08.72 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
ghost_1    | ghost 11:39:08.76 INFO  ==> Ensuring Ghost directories exist
ghost_1    | ghost 11:39:08.77 INFO  ==> Trying to connect to the database server
mariadb_1  | mariadb 11:39:10.43 INFO  ==> Starting mariadb in background
mariadb_1  | mariadb 11:39:12.46 INFO  ==> Configuring authentication
mariadb_1  | mariadb 11:39:12.56 INFO  ==> Running mysql_upgrade
mariadb_1  | mariadb 11:39:12.92 INFO  ==> Stopping mariadb
mariadb_1  | 
mariadb_1  | mariadb 11:39:18.33 INFO  ==> ** MariaDB setup finished! **
mariadb_1  | mariadb 11:39:18.36 INFO  ==> ** Starting MariaDB **
mariadb_1  | 2021-07-19 11:39:18 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.3.30-MariaDB) starting as process 1 ...
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Using Linux native AIO
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Uses event mutexes
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Number of pools: 1
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Using SSE2 crc32 instructions
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Waiting for purge to start
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: 10.3.30 started; log sequence number 1625452; transaction id 20
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2021-07-19 11:39:18 0 [Note] InnoDB: Buffer pool(s) load completed at 210719 11:39:18
mariadb_1  | 2021-07-19 11:39:18 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1  | 2021-07-19 11:39:18 0 [Note] Server socket created on IP: '0.0.0.0'.
mariadb_1  | 2021-07-19 11:39:18 0 [Warning] 'proxies_priv' entry '@% root@0d5f08595874' ignored in --skip-name-resolve mode.
mariadb_1  | 2021-07-19 11:39:18 0 [Note] Reading of all Master_info entries succeeded
mariadb_1  | 2021-07-19 11:39:18 0 [Note] Added new Master_info '' to hash table
mariadb_1  | 2021-07-19 11:39:18 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1  | Version: '10.3.30-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
ghost_1    | ghost 11:39:18.81 INFO  ==> Configuring database
ghost_1    | ghost 11:39:18.82 INFO  ==> Setting up Ghost
ghost_1    | ghost 11:39:19.81 INFO  ==> Configuring Ghost URL to http://localhost
ghost_1    | ghost 11:39:19.82 INFO  ==> Passing admin user creation wizard
ghost_1    | ghost 11:39:19.92 INFO  ==> Starting Ghost in background
ghost_1    | /opt/bitnami/scripts/libghost.sh: line 91: /opt/bitnami/ghost/content/logs/ghost.log: No such file or directory
ghost-compose_ghost_1 exited with code 1
```
It seems like I still don't have a handle on the file/folder permissions.

**Comment by javsalgar** [comment]
Could you confirm that the volumes are empty before starting the Ghost instance? Maybe there are leftovers from the previous installations.

**Comment by jeffreycunn** [comment]
> Could you confirm that the volumes are empty before starting the Ghost instance? Maybe there are leftovers from the previous installations.

Yes, I confirmed the volumes are empty.  On several attempts, I pre-built the sub-directories for the mounts (left empty) and then on several occasions, I had Docker make the mount sub-directories. Neither scenario worked.

**Comment by jeffreycunn** [comment]
I recently tested this on a GCP Debian host and had success. My first host was an Ubuntu OS and so I am wondering if that makes a difference for some reason.

**Comment by jeffreycunn** [comment]
I think my first success was coincidental to starting with a fresh install, not necessarily because I used a VM with Debian OS.

The problem popped up again. I think it is related to the removal of the ./content directory  within the ./ghost_data directory.  That directory needs to be in place apparently?

What I think would help is to know which directories and files should exist when restarting the containers.  It's important to ensure all of the files and directories have the right owners and permissions set clearly...  I was able to get the ghost and mariadb containers running again, but after much trial and error.  I am still slightly confused on permissions, etc. and knowing how to properly inject changes to the config.production.json.
